doc/reference/storage_powerflex.md

```diff
@@ -0,0 +1,151 @@
+For storage volumes with content type `filesystem` (containers and custom file-system volumes), the `powerflex` driver uses volumes with a file system on top (see [`block.filesystem`](storage-powerflex-vol-config)).
+By default, LXD creates thin-provisioned PowerFlex volumes.
+
+LXD expects the PowerFlex protection domain and storage pool already to be setup.
+Furthermore LXD assumes that it has full control over the storage pool.
+Therefore, you should never maintain any volumes that are not owned by LXD in a PowerFlex storage pool, because LXD might delete them.
+
+This driver also behaves differently than other drivers in that it provides remote storage.
+As a result and depending on the internal network, storage access might be a bit slower than for local storage.
+On the other hand, using remote storage has big advantages in a cluster setup, because all cluster members have access to the same storage pools with the exact same contents, without the need to synchronize storage pools.
+
+When creating a new storage pool using the `powerflex` driver, LXD tries to discover one of the SDT from the given storage pool.
+This SDT can be overwritten using [`powerflex.sdt`](storage-powerflex-pool-config).
+LXD instructs the NVMe initiator to connect to all the other SDT when first connecting to the subsystem.
+
+Due to the way copy-on-write works in PowerFlex, snapshots of any volume don't rely on its parent.
+As a result, volume snapshots are fully functional volumes themselves from whom additional snapshots can be taken.
+This tree of dependencies is called the PowerFlex vTree.
+Both volumes and their snapshots get added as standalone NVMe disks to the LXD host.
+
+(storage-powerflex-volume-names)=
+### Volume names
+
+Due to a [limitation](storage-powerflex-limitations) in PowerFlex, volume names cannot exceed 31 characters.
+In order for LXD to identify the volumes type and snapshots, special identifiers are appended to the volumes name:
+
+Type            | Identifier   | Example
+:--             | :---         | :----------
+Container       | `c_`         | `c(_project1)_c1`
+Virtual machine | `v_`         | `v(_project2)_v1`
+Image           | `i_`         | `i(_project3)_img`
+Custom volume   | `u_`         | `u(_project4)_vol`
+
+When creating volumes in the `default` project its identifier gets omitted.
+This does not apply for custom storage volumes since their name may contain an underscore which could
+potentially lead to inconsistencies when running the `lxd recover` command.
+
+Volumes for virtual machines and ISO files require another two characters.
+The VMs root volume contains the `.b` block suffix whilst an ISO file ends on `.i`.
+
+In order to have enough space for the snapshot identifier, at least two more characters need to be
+reserved. You can try keeping the snapshots name at a minimum by setting [`snapshots.pattern`](storage-powerflex-snapshot-pattern)
+to `%d`. This will result in the following volume names:
```

Comment:
Again, we should specify what that actually means for the user (like "So if you plan on taking snapshots, don't use instance or volume names that exceed xx characters").

Reply:
I'll add another sentence that depicts that explicitly.